src/dtgtk/thumbtable.c

```diff
@@ -2526,57 +2526,79 @@ static gboolean _filemanager_key_move(dt_thumbtable_t *table, dt_thumbtable_move
+                                &stmt, NULL);
+    if(sqlite3_step(stmt) == SQLITE_ROW) maxrowid = sqlite3_column_int(stmt, 0);
+    sqlite3_finalize(stmt);
 
-  int newrowid = baserowid;
-  // last rowid of the current collection
-  int maxrowid = 1;
-  sqlite3_stmt *stmt;
-  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
-                              "SELECT MAX(rowid) FROM memory.collected_images", -1,
-                              &stmt, NULL);
-  if(sqlite3_step(stmt) == SQLITE_ROW) maxrowid = sqlite3_column_int(stmt, 0);
-  sqlite3_finalize(stmt);
+    switch(move)
+    {
+      // classic keys
+      case DT_THUMBTABLE_MOVE_LEFT:
+        newrowid = MAX(baserowid - 1, 1);
+        break;
+      case DT_THUMBTABLE_MOVE_RIGHT:
+        newrowid = MIN(baserowid + 1, maxrowid);
+        break;
+      case DT_THUMBTABLE_MOVE_UP:
+        newrowid = MAX(baserowid - table->thumbs_per_row, 1);
+        break;
+      case DT_THUMBTABLE_MOVE_DOWN:
+        newrowid = MIN(baserowid + table->thumbs_per_row, maxrowid);
+        break;
 
-  // classic keys
-  if(move == DT_THUMBTABLE_MOVE_LEFT && baserowid > 1)
-    newrowid = baserowid - 1;
-  else if(move == DT_THUMBTABLE_MOVE_RIGHT && baserowid < maxrowid)
-    newrowid = baserowid + 1;
-  else if(move == DT_THUMBTABLE_MOVE_UP && baserowid - table->thumbs_per_row >= 1)
-    newrowid = baserowid - table->thumbs_per_row;
-  else if(move == DT_THUMBTABLE_MOVE_DOWN && baserowid + table->thumbs_per_row <= maxrowid)
-    newrowid = baserowid + table->thumbs_per_row;
-  // page key
-  else if(move == DT_THUMBTABLE_MOVE_PAGEUP)
-  {
-    newrowid = baserowid - table->thumbs_per_row * (table->rows - 1);
-    while(newrowid < 2 - table->thumbs_per_row) newrowid += table->thumbs_per_row;
-  }
-  else if(move == DT_THUMBTABLE_MOVE_PAGEDOWN)
-  {
-    newrowid = baserowid + table->thumbs_per_row * (table->rows - 1);
-    while(newrowid > maxrowid) newrowid -= table->thumbs_per_row;
+      // page keys
+      case DT_THUMBTABLE_MOVE_PAGEUP:
+        newrowid = baserowid - table->thumbs_per_row * (table->rows - 1);
+        while(newrowid < 1) newrowid += table->thumbs_per_row;
+        if(newrowid == baserowid) newrowid=1;
+        break;
+      case DT_THUMBTABLE_MOVE_PAGEDOWN:
+        newrowid = baserowid + table->thumbs_per_row * (table->rows - 1);
+        while(newrowid > maxrowid) newrowid -= table->thumbs_per_row;
+        if(newrowid == baserowid) newrowid=maxrowid;
+        break;
+
+      // direct start/end
+      case DT_THUMBTABLE_MOVE_START:
+        newrowid = 1;
+        break;
+      case DT_THUMBTABLE_MOVE_END:
+        newrowid = maxrowid;
+        break;
+      default:
+        break;
+    }
   }
-  // direct start/end
-  else if(move == DT_THUMBTABLE_MOVE_START)
-    newrowid = 1;
-  else if(move == DT_THUMBTABLE_MOVE_END)
-    newrowid = maxrowid;
 
-  if(newrowid == baserowid) return FALSE;
+  //if(newrowid == baserowid) return FALSE;
```

Comment:
is that the "fix" part of your PR (I don't reproduce your "cursor" hiding with master)
Anyway, if that's the case, better remove the line for real. git will keep the memory of it if needed !

Reply:
No, looks like I broke something in git 😅. Will clean it up